concurrency.go

```diff
@@ -117,10 +127,12 @@ func WaitFor(condition func(i int) bool, maxDuration time.Duration, tick time.Du
 
 	for {
 		select {
+		case <-ctx.Done():
+			return i, time.Since(start), false
 		case <-timer.C:
 			return i, time.Since(start), false
```

Comment:
My current implementation is naive.

I will rewrite to use 

ctx, cancel := context.WithTimeout(ctx, maxDuration)

Then no ticker is needed, and the implantation is safer and simpler

Reply:
letz go